tests/compute/entry-point-uniform-params.slang

```diff
@@ -25,16 +27,17 @@ struct Things
 // A shader parameter at global scope should be assigned
 // a register/binding before any related to the entry point.
 
-//TEST_INPUT:cbuffer(data=[1 0 0 0]):dxbinding(0),glbinding(0)
+//TEST_INPUT:cbuffer(data=[1 0 0 0]):dxbinding(0),glbinding(0),name=signs
 ConstantBuffer<Signs> signs;
 
 [numthreads(4, 1, 1)]
 void computeMain(
-//TEST_INPUT:cbuffer(data=[2 0 0 0 3 0 0 0]):dxbinding(1),glbinding(1)
+//TEST_INPUT:cbuffer(data=[2 0 0 0 3 0 0 0]):dxbinding(1),glbinding(1),name=stuff
 	uniform Stuff  stuff,
+//TEST_INPUT:cbuffer(data=[3]):,isCPUOnly,name=things    
```

Comment:
I can't remember if I already chimed in on the `isCPUOnly` thing and the lack of a matching layout between CPU and GPU targets for the `data=[...]` stuff.

My thought is that for 99% of cases the right fix would be that we should use name and reflection-based parameter setting for all targets (not just CPU) to avoid this problem. This would mean that `data=...` should probably accept some kind of structured input representation, perhaps even resembling C-style initializer lists. Struct fields and array elements would then be set based on their declaration order, independent of memory layout.

(Eventually we'd want that kind of thing to be based on initializers expressed in Slang directly, that the user can query via reflection, but there are a ton of design questions about what an API for querying default parameter values should look like)

The main reason not to switch to reflection-based parameter setting (other than the cost of implementing the feature and refactoring that many tests) is that we still have some tests that compare against running shaders compiled via fxc/dxc/glslang, which would preclude using reflection for parameter binding (unless we added coded to set parameters via their reflection interfaces as well). The Right Thing is probably to support both raw-data-based setting and reflection-based setting, but favor reflection.